### ISSUE TITLE
Electron browser window gets white screen in sandbox mode

### ISSUE THREAD

**Issue by Alexandra1337** [electron/electron#10511]
I have electron with react app
```
electron: 1.6.11
electron-packager: 8.7.2
react: 15.4.2
```
I create my build with electron-packager
```
electron-packager . --ignore=./.gitignore --ignore=./.node_modules --ignore=./webpack.config.js --platform=mas --arch=all --icon='./electron.icns' --app-bundle-id='com.my.id' --helper-bundle-id='com.my.id.helper' --app-category-type=public.app-category.utilities --overwrite --asar=true --app-version='1.0.4' --out=release-builds
```
The build works fine for now, after that I sign the app with script
```
#!/bin/bash

# Name of your app.
APP="SmarterBack"
# The path of your app to sign.
APP_PATH="release-builds/SmarterBack-mas-x64/SmarterBack.app"
# APP_PATH="release-builds/SmarterBack-darwin-x64/SmarterBack.app"
# The path to the location you want to put the signed package.
RESULT_PATH="/Users/admin/Desktop/$APP.pkg"
# The name of certificates you requested.
APP_KEY="3rd Party Mac Developer Application: JENGO LLC (63UV74H5Q5)"
INSTALLER_KEY="3rd Party Mac Developer Installer: JENGO LLC (63UV74H5Q5)"
# The path of your plist files.
CHILD_PLIST="child.plist"
PARENT_PLIST="parent.plist"

FRAMEWORKS_PATH="$APP_PATH/Contents/Frameworks"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/Electron Framework.framework/Versions/A/Electron Framework"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/Electron Framework.framework/Versions/A/Libraries/libffmpeg.dylib"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/Electron Framework.framework/Versions/A/Libraries/libnode.dylib"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/Electron Framework.framework"

codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/$APP Helper.app/Contents/MacOS/$APP Helper"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/$APP Helper.app/"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/$APP Helper EH.app/Contents/MacOS/$APP Helper EH"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/$APP Helper EH.app/"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/$APP Helper NP.app/Contents/MacOS/$APP Helper NP"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$FRAMEWORKS_PATH/$APP Helper NP.app/"
codesign --entitlements "$CHILD_PLIST" -v -f -s "$APP_KEY" "$APP_PATH/Contents/MacOS/$APP"

codesign --entitlements "$PARENT_PLIST" -v -f -s "$APP_KEY" "$APP_PATH"
```
And after i sign my app I start to get blank screen on my browser window and nothing can load. After some debugging I notice that I can't use require to import my js, css in my index.html . I started to research more and I found that I had to use "sandbox: true" in my window
```
window = new BrowserWindow({
    webPreferences: {
        sandbox: true,
        nodeIntegration: false
    }
})
```

I even tried to use preload.js to the webPreferences
```
var hostProcess = process;
var hostRequire = require;

process.once('loaded', function(){

  global.host = {
    process: hostProcess,
    crash: function () {
      hostProcess.crash();
    },
  };

  global.host.ipc = hostRequire('ipc');
  global.host.ipc = hostRequire('fs');
});
```

My Index.html
```
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="Content-Security-Policy" content="default-src 'none';script-src 'self'">
    <title>SmarterBack Preferences</title>
    <link rel="stylesheet" type="text/css" href="bootstrap.min.css">
    <link rel="stylesheet" type="text/css" href="quill.snow.css">
    <link rel="stylesheet" type="text/css" href="Main.css">
</head>

<body>
    <div id="main"></div>
</body>

<script src="js/libs/createjs/preloadjs-0.6.2.min.js"></script> 
<script src="js/BRFv4Demo.js"></script>
<script src="js/jquery.min.js"></script>
<script src="js/jquery-3.2.1.min.js"></script>
<script src="js/bootstrap.min.js"></script>
<script type="text/javascript" src="/Main.js"></script>
</html>
```
For now nothing seems to work. When the app is in sandbox mode I can't use any node processes and my html can't require anything so I just end up with one blank screen.I've seen many posts with problem like this one, but haven't found a solution to my problem yet.

**Comment by Alexandra1337** [comment]
Any kind of advice or help would be appreciated.

**Comment by 3webbg** [comment]
I have the same problem, some solution?

**Comment by sethlu** [comment]
Issue closed in electron-userland/electron-osx-sign#151
